README.md

```diff
@@ -7,290 +7,505 @@
+- **guzzle_options**: Configuration propagated to Guzzle when fetching the JWKs (RS256 only). These options are [documented here](http://docs.guzzlephp.org/en/stable/request-options.html).
+- **secret\_base64\_encoded**: When `true`, it will decode the secret used to verify the token signature. This is only used for HS256 tokens and defaults to `true`. Your Application settings will say whether the Client Secret provided is encoded or not. 
+
+## Usage - Management API
+
+This SDK also provides a wrapper for the Management API, which is used to perform operations on your Auth0 tenant. Using this API, you can:
+
+- Search for and create users
+- Create and update Applications
+- Retrieve log entries
+- Manage rules 
+
+... and much more. See our [documentation](https://auth0.com/docs/api/management/v2) for information on what's possible and the examples below for how to authenticate and access this API. 
+
+### Authentication
+
+In order to use the Management API, you must authenticate one of two ways:
+
+- For temporary access or testing, you can [manually generate an API token](https://auth0.com/docs/api/management/v2/tokens#get-a-token-manually) and save it in your `.env` file
+- For extended access, you can create and execute and Client Credentials grant (detailed above) when access is required
+
+Regardless of the method, the token generated must have the scopes required for the operations your app wants to execute. Consult the [API documentation](https://auth0.com/docs/api/management/v2) for the scopes required for the specific endpoint you're trying to access.
+
+To grant the scopes needed: 
+
+1. Go to [APIs](https://manage.auth0.com/#/apis) > Auth0 Management API > **Machine to Machine Applications** tab.
+2. Find your Application and authorize it.
+3. Click the arrow to expand the row and select the scopes required.
+
+Now you can authenticate one of the two ways above and use that token to perform operations:
+
+```php
+use Auth0\SDK\API\Management;
+
+$access_token = getenv('AUTH0_MANAGEMENT_API_TOKEN');
+if ( empty( $access_token ) ) {
+	// See "Client Credentials Grant" above
+	$access_token = get_access_token();
+}
+$mgmt_api = new Management( $access_token, getenv('AUTH0_DOMAIN') );
 ```
 
-You will get your app client and secret from your Auth0 app you had created.
-The auth0 domain, is the one you pick when you created your auth0 account.
-You need to set this callback url in your app allowed callbacks.
-The app token is used in the 'create user' page and needs `create:users` scope. To create one, you need to use the token generator in the [API V2 documentation page](https://auth0.com/docs/apiv2)
+The `Management` class stores access to endpoints as properties of its instances. The best way to see what endpoints are covered is to read through the `\Auth0\SDK\API\Management::__construct()` method. 
```

Comment:
I'm not saying you're lazy for not listing them, but a short list would probably be better. Check how I'm doing that in [java](https://github.com/auth0/auth0-java#management-api). This comment is tied to the "official API documentation" one above

Reply:
Thank you for not saying I'm lazy ... this already took more than a day (not including review comments 😆 )!

I need to generate a list of what's covered but I don't think it's essential for this re-write. I have this task captured on my end.